src/pagination/index.tsx

```diff
@@ -0,0 +1,99 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import i18n from '@dojo/framework/core/middleware/i18n';
+import theme from '@dojo/framework/core/middleware/theme';
+
+import * as css from '../theme/default/pagination.m.css';
+import { WNode } from '@dojo/framework/core/interfaces';
+import Icon from '../icon';
+
+export interface PaginationProperties {
+	initialPage?: number;
+	total: number;
+	onPageChange(page: number): void;
+}
+
+interface PaginationCache {
+	currentPage: number;
+}
+
+const icache = createICacheMiddleware<PaginationCache>();
+const factory = create({ theme, icache, i18n }).properties<PaginationProperties>();
+
+const PAGE_LINKS = 3;
```

Comment:
Do you think this should be customisable? 3 seems a little small, especially if the widget is given a lot more space. Could you use `dimensions` / `resize` middleware perhaps to react to the space available and render what will fit?

Reply:
Everything is possible. ;)

Reply:
Maybe up to some maximum?

Reply:
Moved off to issue #1239